source/opt/loop_dependence.cpp

```diff
@@ -817,6 +872,175 @@ bool LoopDependenceAnalysis::WeakCrossingSIVTest(
   return false;
 }
 
+// Calculate the greatest common divisor of a & b using Stein's algorithm.
+// https://en.wikipedia.org/wiki/Binary_GCD_algorithm
+int64_t GreatestCommonDivisor(int64_t a, int64_t b) {
```

Comment:
Inside an anonymous namespace, please.  Likewise for all the other free functions in this file.  If the functions need to be shared across other files, perhaps make the class static?

Reply:
I've left them in the anonymous namespace, as they are only used here.